presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java

```diff
@@ -52,17 +52,34 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
 
     private static String formatErrorMessage(OperatorType operatorType, List<? extends Type> argumentTypes, Optional<TypeSignature> returnType)
     {
-        String operatorString;
         switch (operatorType) {
+            case ADD:
+            case SUBTRACT:
+            case MULTIPLY:
+            case DIVIDE:
+            case MODULUS:
+            case EQUAL:
+            case NOT_EQUAL:
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                return format("Cannot apply operator: %s %s %s", argumentTypes.get(0), operatorType.getOperator(), argumentTypes.get(1));
+            case NEGATION:
+                return format("Cannot negate %s", argumentTypes.get(0));
             case BETWEEN:
                 return format("Cannot check if %s is BETWEEN %s and %s", argumentTypes.get(0), argumentTypes.get(1), argumentTypes.get(2));
+            case IS_DISTINCT_FROM:
+                return format("Cannot check if %s IS DISTINCT FROM %s", argumentTypes.get(0), argumentTypes.get(1));
             case CAST:
-                operatorString = format("%s%s", operatorType.getOperator(), returnType.map(value -> " to " + value).orElse(""));
-                break;
-            default:
-                operatorString = format("'%s'%s", operatorType.getOperator(), returnType.map(value -> ":" + value).orElse(""));
+                return format("Cannot cast %s to %s", argumentTypes.get(0), returnType.orElseThrow());
+            case SUBSCRIPT:
+                return format("Cannot use %s for subscript of %s", argumentTypes.get(1), argumentTypes.get(0));
         }
-        return format("%s cannot be applied to %s", operatorString, Joiner.on(", ").join(argumentTypes));
+        return format("Operator '%s'%s cannot be applied to %s",
+                operatorType.getOperator(),
+                returnType.map(value -> ":" + value).orElse(""),
```

Comment:
Do we have a test case which would hit the `returnType.isPresent()` case?

Reply:
I'm not aware of a way for a query to hit any of the other operators here.